nazgul/mysql_model.py

```diff
@@ -1,218 +1,156 @@
 from flask import Flask, render_template, request
 import mysql.connector
 
-try:
-    import nazgul.xmlrenderer as xmlrenderer
-except:
-    import xmlrenderer
-
 
 class MySQLModel:
     def __init__(self, user="root", host="127.0.0.1", password="", db="bouncer"):
+        self._pass = password
         self._user = user
         self._host = host
-        self._pass = password
-        self._db = db
+        self._db_name = db
+        # TODO: Should I have autocommit always True?
+        self._db = mysql.connector.connect(
+            user=user, host=host, database=db, autocommit=True
+        )
 
     def index(self):
         return "Welcome to Nazgul"
 
     def location_show(self, product, fuzzy):
-        xml = xmlrenderer.XMLRenderer()
-
-        if product is None:
-            return xml.error("The GET parameter product is required", errno=103), False
         if fuzzy:
             sql = """SELECT distinct id FROM mirror_products mp WHERE name LIKE %s;"""
             product = "%" + product + "%"
         else:
             sql = """SELECT distinct id FROM mirror_products WHERE name = %s;"""
 
-        db = mysql.connector.connect(user="root", host=self._host, database=self._db)
-        cur = db.cursor()
+        cur = self._db.cursor()
         cur.execute(sql, (product,))
 
         res = cur.fetchall()
 
         cur.close()
-        db.close()
 
         ids = []
         for line in res:
             ids.append(line[0])
 
         locs = self.get_locations_info(ids)
 
-        for loc in locs:
-            xml.prepare_locations(loc)
-        return xml.render(), True
+        return locs, True
 
     def location_add(self, product, os, path):
-        xml = xmlrenderer.XMLRenderer()
-
-        if not (product and os and path):
-            return (
-                xml.error(
-                    "product, os, and path are required POST parameters.", errno=101
-                ),
-                False,
-            )
-
         os_exists, os_id = self.os_exists(os)
         product_exists, product_id = self.product_exists(product)
         location_exists, location_id = self.location_exists(os, product)
 
         if not os_exists:
-            return xml.error("FAILED: '" + os + "' does not exist", errno=104), False
+            return 106, False
```

Comment:
Returning a number makes it a bit hard to reason about. We should keep the errnos to maintain backward capability, but it might be better to create a new exception type with `msg` and `errno` fields, then throw it on error. When those are caught in the view, you can just return the 400 with `xml.error(e.msg, e.errno)` that way the explanation is with the code checking, but the display logic is still in the view.